externals/llvm-external-projects/torch-mlir-dialects/include/torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.td

```diff
@@ -514,4 +514,28 @@ def Tcp_Atan2Op : Tcp_BinaryElementwiseOp<"atan2", [SameOperandsAndResultElement
   let assemblyFormat = "$in1 `,` $in2 attr-dict `:` type($in1) `,` type($in2) `->` type($out)";
 }
 
+def Tcp_CastOp:  Tcp_Op<"cast", [Pure, Elementwise, SameOperandsAndResultShape]> {
+
+  let summary = "TCP Cast operation";
+
+  let description = [{
+    Performs elementwise type conversion.
+  }];
+
+  let arguments = (ins
+    Tcp_Tensor:$in,
+    // Align with enum SignednessSemantics
```

Comment:
Is there a better way to handle attributes that are enum?

Reply:
@burmako What is the recommended approach for defining attributes that represent enums like this?

Reply:
Here's how we're doing this in StableHLO:
  * Definition of an enum attribute: https://github.com/openxla/stablehlo/blob/d67da5d85045e03893684c35ae5e730695289ae2/stablehlo/dialect/StablehloEnums.td#L98-L122.
  * Use of this attribute: https://github.com/openxla/stablehlo/blob/d67da5d85045e03893684c35ae5e730695289ae2/stablehlo/dialect/StablehloOps.td#L1570.

Reply:
Thanks a lot for those pointers @burmako. Thats very helpful.

@zezhang Can we follow a similar approach to represent signedness in `tcp.cast`?